rfcs/rfcOVID-0019.md

```diff
@@ -0,0 +1,329 @@
+revisit this decision in the future.
+
+## Backwards Compatibility
+
+For basic usage without an `import` method, there are no
+backwards-compatibility issues. There are, however a few surprises.
+
+* Debugger
+
+When using the debugger, it's common to test the return value of a subroutine:
+
+```
+auto(-1)  DB<2> v
+30
+31:    if (@errors) {
+32:        warn join "\n" => @errors;
+33==>    say status();
+34     }
+35
+36:    unless ( $opt_for{description} ) {
+37:        say "Please enter a description for this $opt_for{type} entry:";
+38:        chomp( my $description = <STDIN> );
+39:        $opt_for{description} = $description;
+DB<2> x status()
+Undefined subroutine &main::status called at ...
+```
+
+The subroutine is bound a compile-time and removed from the namespace. You can
+step into it, but you can't call it directly from the debugger. This should be
+considered.
+
+* Exporting Spec
+
+We often have the following near the top of our code:
+
+```perl
+our @EXPORT    = qw(quux);
+our @EXPORT_OK = qw(
+    bar
+    baz
+    bay
+)
+our %EXPORT_TAGS = qw(
+    'all'     => \@EXPORT_OK,
+    'strings' => qw(bay baz),
+    'other'   => qw(bay)
+);
+```
+
+That would be rewritten as this:
+
+```perl
+sub quux :export(DEFAULT)       {...}
+sub bar  :export                {...}
+sub baz  :export(strings)       {...}
+sub bay  :export(strings,other) {...}
+```
+
+Some developers like a single exporting spec at the top of a module. Others
+like viewing a function and seeing directly if it can be exported:
+
+```perl
+sub foo :export () {
+    ...
+}
+```
+
+* The `import()` method
+
+This one is tricky because we have to maintain backwards-compatibility.
+
+If there is no `import()` method defined in the class  then any attempt to
+import a function not designated for export should be a fatal error:
+
+```
+$ perl -MMy::Module=whee -E 1
+"whee" is not exported by the My::Module module
+Can't continue after import errors at -e line 0.
+BEGIN failed--compilation aborted.
+```
+
+Any attempt to import a non-existent export (e.g., `use Foo ':bar'`) should
+also be fatal.
+
+For the following, we'll use the following stub example of an `import()`
+method:
+
+```perl
+package My::Module {
+    use v5.38;
+
+    sub import ($class, @args) {
+        ...
+    }
+    sub bar  :export           {...} # @EXPORT_OK
+    sub baz  :export(strings)  {...} # %EXPORT_TAGS
+    sub bay  :export(strings)  {...} # %EXPORT_TAGS
+    sub quux :export(DEFAULT)  {...} # @EXPORT
+    sub whee                   {...} # not exported
+}
+```
+
+If an `import()` method is defined in the package, it will need to call
+`UNIVERSAL::import()` to have the lexical importing handled correctly.
+
+```perl
+sub import ($class, @args) {
+    my ( $import, $mine ) = separate_args(@args);
+    # do stuff with $mine
+    UNIVERSAL::import($import->@*);
+}
+```
+
+For the above, how do we tell `UNIVERSAL::import` which class its importing
```

Comment:
The correct way to call this would be `$class->UNIVERSAL::import(...)`. This is the same as wrapping any other method. Since this is doing lexical exports, there is no package being imported into. `caller` does not need to be maintained or communicated.

Reply:
Fixed.